### PR TITLE
Housekeeping: remove data-automation-id

### PR DESCRIPTION
**Pull request by gyfchong**
## Why
No longer support `data-automation-id`

## What
Remove tests etc. that mention it.

### CODE REVIEW

File: packages/components/src/Heading/Heading.spec.tsx [hunk 1]
```diff
@@ -42,14 +42,12 @@ describe("<Heading />", () => {
...
-      <Heading variant="display-0" data-automation-id="test-id">
+    const { getByTestId } = render(
+      <Heading variant="display-0" data-testid="test-id">
         Example
       </Heading>
     )
-    expect(
-      container.querySelector('[data-automation-id="test-id"]')
-    ).not.toBeNull()
+    expect(getByTestId("test-id")).not.toBeNull()
```
Comment (HeartSquared) [inline packages/components/src/Heading/Heading.spec.tsx:50]:
```suggestion
    expect(getByTestId("test-id")).toBeInTheDocument()
```

File: packages/components/src/Text/Text.spec.tsx [hunk 2]
```diff
@@ -70,14 +70,12 @@ describe("<Text />", () => {
...
-      <Text variant="intro-lede" data-automation-id="test-id">
+    const { getByTestId } = render(
+      <Text variant="intro-lede" data-testid="test-id">
         Example
       </Text>
     )
-    expect(
-      container.querySelector('[data-automation-id="test-id"]')
-    ).not.toBeNull()
+    expect(getByTestId("test-id")).not.toBeNull()
```
Comment (HeartSquared) [inline packages/components/src/Text/Text.spec.tsx:78]:
```suggestion
    expect(getByTestId("test-id")).toBeInTheDocument()
```

File: packages/design-tokens/docs/DocsComponents.tsx [hunk 3]
```diff
@@ -70,7 +70,6 @@ const TabbedCodeBlocks = ({
               // eslint-disable-next-line jsx-a11y/anchor-is-valid
               <a
                 style={{ flexShrink: 0 }}
-                data-automation-id={tab.automationId}
```
Comment (HeartSquared) [inline packages/design-tokens/docs/DocsComponents.tsx:73]: Do we need to remove the `automationId` prop then? Or put it into a `data-testid`?

Comment (gyfchong) [reply packages/design-tokens/docs/DocsComponents.tsx:73]: Definitely probably don't need this

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**